shared/js/background/classes/tab.es6.js

```diff
@@ -70,6 +70,14 @@ class Tab {
         return this._tabState.adClick
     }
 
+    set disabledClickToLoadRuleActions (value) {
+        this._tabState.setValue('disabledClickToLoadRuleActions', value)
+    }
+
+    get disabledClickToLoadRuleActions () {
+        return this._tabState.disabledClickToLoadRuleActions
+    }
```

Comment:
I think you're doing too much work just to have `disabledClickToLoadRuleActions` serializable. As you have the `Tab` wrapping the `_tabState`, can't you leverage accessors and class methods to make it seem like a set, but have a list in the background?

i.e.
Tabstate:
```javascript
/** @type {string[]} */
this.disabledClickToLoadRuleActions = []
```
Tab:
```javascript
get disabledClickToLoadRuleActions () {
   return new Set(this._tabState.disabledClickToLoadRuleActions)

// or an explicit action checker function
isClickToLoadRuleActionDisabled (action) {
  return this._tabState.disabledClickToLoadRuleActions.contains(action)
}

addDisabledClickToLoadRuleAction (action) {
  if (!this._tabState.disabledClickToLoadRuleActions.contains(action)) {
     this._tabState.disabledClickToLoadRuleActions.push(action)
  }
}
```

Reply:
I like the idea of having set/get wrap the Array in a Set, but how will that work with saving mutations. I don't know how this tabState storage magic works, but I'm assuming the mutations made to a Set based on the Array in tabState might be lost?

I don't mind using `Array.prototype.contains` when adding disabled rule actions, but I'm against using that when checking if a rule action is disabled - that happens quite often.

I'm struggling to see how to apply your suggestion with the above constraints, but I'm open to doing this differently. I don't particularly like having the Record<string, null> Object hack either. What do you think?

Reply:
What is the concern with `Array.prototype.contains`? My assumption was that `disabledClickToLoadRuleActions` would by mostly an empty list, and then rarely 1 or 2 elements long. In that case, then the overhead of using a `Set` is probably higher than just looking through the array. If that assumption is wrong, then maybe I need to reread the rest of this PR to understand how this works.

Reply:
Yea, fair enough, you're right that it will usually be an empty list and even when not the list will be short. The lookup will be cheap enough either way.

I started making the change to use an Array instead, but I do wonder, is it worth it? It seems the change is basically that we can have

```javascript
        getDefaultEnabledClickToLoadRuleActionsForTab(thisTab).filter(
            ruleAction => !disabledClickToLoadRuleActions.includes(ruleAction)
        )
```

instead of

```javascript
        getDefaultEnabledClickToLoadRuleActionsForTab(thisTab).filter(ruleAction =>
            !Object.prototype.hasOwnProperty.call(
                thisTab.disabledClickToLoadRuleActions,
                ruleAction
            )
        )
```

in before-request.es6.js.

At the expense of needing to having to have

```javascript
    if (!tab.disabledClickToLoadRuleActions.includes(data.action)) {
        tab.disabledClickToLoadRuleActions.push(data.action)
    }
```

instead of

```javascript
    tab.disabledClickToLoadRuleActions[data.action] = null
```

in message-handlers.js.

Reply:
I don't feel too strongly about it either way, so I've changed it now anyway. Done.

Reply:
Thanks - I think the readability gain is worth it.